src/ripple/protocol/Quality.h

```diff
@@ -200,6 +200,29 @@ class Quality
             toAmount<In>(stRes.in), toAmount<Out>(stRes.out));
     }
 
+    Amounts
+    ceil_in_strict(Amounts const& amount, STAmount const& limit, bool roundUp)
+        const;
+
+    template <class In, class Out>
+    TAmounts<In, Out>
+    ceil_in_strict(
+        TAmounts<In, Out> const& amount,
+        In const& limit,
+        bool roundUp) const
+    {
```

Comment:
This looks like a copy of `ceil_in` with one function called changed. 

If `ceil_in` is being deprecated and will be replaced with `ceil_in_strict` then I actually prefer the code duplication. And I think this is the case. If this is not the case, then let's look at merging the two.

On the same topic, if the `ceil_in` is being deprecated, consider renaming `ceil_in` to `ceil_in_deprecated` or somesuch and renaming `ceil_in_strict` to `ceil_in`.

Reply:
I'm not sure about `ceil_in` being deprecated.  But it certainly has an odd behavior regarding rounding.  So we _might_ eventually want to replace all uses of it with `ceil_in_strict`.  But I'd like to hold off on renaming either of them until we know that we're actually going to make that change (which would have to go under an amendment).  There are a few things in this code base that have been "deprecated" for quite a few years without ever being removed.  I'd like to not add to that confusion.